google/cloud/security/common/data_access/group_dao.py

```diff
@@ -90,3 +108,73 @@ def get_recursive_members_of_group(self, group_email, timestamp):
                 if member.get('member_type') == 'GROUP':
                     queue.put(member.get('member_id'))
         return all_members
+
+    def get_recursive_members(self, node, timestamp):
+        """Get all the recursive members of a group.
+
+        Args:
+            node: Member node from which to start getting the recursive
+                members.
+            timestamp: String of snapshot timestamp, formatted as
+                YYYYMMDDTHHMMSSZ.
+
+        Returns:
+            node: Member node with all its recursive members.
+        """
+        queue = Queue()
+        queue.put(node)
+        
+        while not queue.empty():
+            node = queue.get()
+            members = self.get_group_members('group_members', node.member_id,
+                                             timestamp)
+
+            for member in members:
+                member_node = MemberNode(member.get('member_id'),
+                                         member.get('member_email'),
+                                         member.get('member_type'),
+                                         member.get('member_satus'),
+                                         node)
+                if member_node.member_type == 'GROUP':
+                    queue.put(member_node)
+        
+        return node
+
+    def build_group_tree(self, timestamp):
+        """Build a tree of all the groups in the organization.
+
+        Args:
+            timestamp: String of snapshot timestamp, formatted as
+                YYYYMMDDTHHMMSSZ.
+
+        Returns:
+            The root node that holds the tree structure of all the groups
+                in the organization.
+        """
+        root = MemberNode('my_customer', 'my_customer', None, None, None)
```

Comment:
Also, is there a reason not to change the MemberNode default arguments values to None such that you're instead calling MemberNode('my_customer', 'my_customer') instead of the current one with the Nones?